### ISSUE TITLE
Intermittent failure of twisted.test.test_iutils.UtilsTestCase.testOutputWithErrorIgnored on Windows

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3239]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3239|
|Type|defect|
|Created|2008-05-16 14:25:29Z|
|Branch|https://github.com/twisted/twisted/tree/iutils-errors-3239|

On the server-2k8-x86-py2.5-select slave,

```
===============================================================================
[ERROR]: twisted.test.test_iutils.UtilsTestCase.testOutputWithErrorIgnored

Traceback (most recent call last):
Failure: twisted.trial.util.DirtyReactorAggregateError: Reactor was unclean.
DelayedCalls: (set twisted.internet.base.DelayedCall.debug = True to debug)
<DelayedCall 60931208 [0.0071427822113s] called=0 cancelled=0 Process._pollEvent()>
===============================================================================
```

<details><summary>Searchable metadata</summary>

```
trac-id__3239 3239
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_iutils_errors_3239 branches/iutils-errors-3239
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1210947929000000 1210947929000000
changetime__1221249560000000 1221249560000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

This test is called `twisted.test.test_iutils.ProcessUtilsTests.test_outputWithErrorIgnored` now.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [24796]) Branching to 'iutils-errors-3239'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

Should be fixed now.  There's a way to wait for the process to end if the deferred `getProcessOutput` returns errbacks, now.  The test uses it.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @exarkun|
|-|-|

OK cool, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [24810]) Merge iutils-errors-3239

Author: exarkun
Reviewer: therve
Fixes: [#3239](https://github.com/twisted/twisted/issues/3239)

Augment the `IOError` with which `getProcessOutput` can fail when
it receives bytes on stderr but is not expecting to.  The exception
now offers a `processEnded` attribute bound to a `Deferred` which
fires when the offending process ends.  The unit tests now use this
attribute in order to wait long enough so that the process does not
cause trial to report an unclean reactor.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
